src/internal/packager2/layout/package.go

```diff
@@ -101,9 +102,12 @@ func LoadFromDir(ctx context.Context, dirPath string, opt PackageLayoutOptions)
 		dirPath: dirPath,
 		Pkg:     pkg,
 	}
-	err = validatePackageIntegrity(pkgLayout, opt.IsPartial)
-	if err != nil {
-		return nil, err
+	// do not validate integrity if inspecting, as all files may not be available.
+	if !opt.Inspect {
+		err = validatePackageIntegrity(pkgLayout, opt.IsPartial)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Would `validatePackageIntegrity` work for inspect if opt.IsPartial is set to true?

Reply:
I will give that a shot. If memory serves me right - I believe the SHA comparison between the aggregate checksum and the checksums file won't match as this isn't a complete package.